source/algorithms/timestep_control.cc

```diff
@@ -94,7 +94,7 @@ TimestepControl::advance()
   // first step.
   if (now_val != start())
     {
-      if (strategy_val == doubling && 2 * s <= tolerance_val)
+      if (strategy_val == doubling)
         s *= 2;
       if (s > max_step_val)
```

Comment:
That can't have been the intent. This means that in each time step, you're going to double the time step length until you eventually hit the end time or the maximal time step size.

I strongly suspect that this strategy was never tested, and has always been irretrievably broken. I don't think your patch actually "fixes" it -- the whole scheme here makes no sense. Can you just remove `Strategies::doubling` altogether? It's not been tested, it doesn't work, and it's likely that nobody uses it, so let's just get rid of it.